File: src/Scraper.java
import com.jaunt.Element;
import com.jaunt.Elements;
import com.jaunt.UserAgent;
import org.json.simple.JSONObject;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.util.Properties;
import java.util.Scanner;


/**
 * @author Christopher Lutz
 */
public class Scraper {
    
    public static void main(String[] args) {
        
        try {
            Properties props = new Properties();
            FileInputStream in = new FileInputStream("config/info.properties");
            props.load(in);
            in.close();
            
            if (!props.containsKey("username") || !props.containsKey("password") || !props.containsKey("url")) {
                System.err.println("Invalid properties!");
                System.exit(1);
            }
            
            int prior = 0;
            
            File cache = new File("config/cache.dat");
            
            if (cache.exists()) {
                
                Scanner sc = new Scanner(cache);
                
                if (sc.hasNextInt()) {
                    prior = sc.nextInt();
                }
                
                sc.close();
            }
            
            UserAgent userAgent = new UserAgent();
            userAgent.visit("http://www.sacs.k12.in.us/site/default.aspx?PageType=14&DomainID=4&PageID=1&ModuleInstanceID=74&ViewID=7070de72-c6ac-43a7-b8eb-103562708ba5&IsMoreExpandedView=True");
            
            Elements links = userAgent.doc.findEvery("<h1 class='ui-article-title'>").findEvery("<a>");
            
            new File("json").mkdir();
            
            int index = 0;
            
            for (Element link : links) {
                
                userAgent.visit(link.getAt("href"));
                
                Element title = userAgent.doc.findFirst("<div class='ui-widget-header'>");
                Element content = userAgent.doc.findFirst("<div class='ui-widget-detail'>");
                
                JSONObject obj = new JSONObject();
                obj.put("status", "publish");
                obj.put("title", title.innerText().trim());
                obj.put("content", content.innerHTML().trim());
                
                FileWriter file = new FileWriter("json/" + (index++) + ".json");
                file.write(obj.toJSONString());
                file.close();
            }
            
            File f = new File("jsontodb-config.yml");
            f.createNewFile();
            PrintWriter pw = new PrintWriter(f);
            pw.println("User: " + props.get("username"));
            pw.println("Pass: " + props.get("password"));
            int num = index - prior;
            if (num > 0) {
                pw.println("Matrix:");
                
                for (int i = 0; i < num; i++) {
                    pw.println("  - Files: json/" + (num - i - 1) + ".json");
                    pw.println("    Command: post");
                    pw.println("    Url: " + props.get("url"));
                }
            }
            pw.close();
            
            cache.createNewFile();
            pw = new PrintWriter(cache);
            pw.println(index);
            pw.close();
            
        } catch (Exception e) {
            
            e.printStackTrace();
        }
    }
}

File: lib/scraper.rb
require 'nokogiri'
require 'open-uri'
require 'json'
require 'safe_yaml'

PAGE_URL = 'http://www.sacs.k12.in.us/site/default.aspx?PageType=14&DomainID=4&PageID=1&ModuleInstanceID=74&ViewID=7070de72-c6ac-43a7-b8eb-103562708ba5&IsMoreExpandedView=True'.freeze

page = Nokogiri::HTML(open(PAGE_URL))
links = page.css('h1.ui-article-title a')

index = 0
Dir.mkdir('json') unless Dir.exist?('json')

links.each do |link|
  subpage = Nokogiri::HTML(open(URI.join(PAGE_URL, link['href'])))
  title = subpage.css('div.ui-widget-header h1 span')[0].text.strip
  text = subpage.css('div.ui-widget-detail')[0].to_s
                .gsub("\r\n", ' ')
                .gsub('src="/', 'src="http://www.sacs.k12.in.us/').strip

  json = { title: title, content: text,
           status: 'publish', 'categories[]' => [7], 'tags[]' => [6,8] }
  json_obj = JSON.pretty_generate json

  File.open("json/#{index}.json", 'w') do |file|
    file.write(json_obj)
  end

  index += 1

  print '.'
end

cache = 0
cache = File.read('config/cache.dat') if File.exist?('config/cache.dat')

if (index - cache) > 0
  info_hash = SafeYAML.load(File.read('config/info.yml'), 'config/info.yml')
  result_yaml = { 'User' => info_hash['User'], 'Pass' => info_hash['Pass'] }
  url = info_hash['Url']
  matrix = []

  (0..(index - cache - 1)).each do |i|
    item = { 'Files' => "json/#{i}.json", 'Command' => 'post', 'Url' => url }
    matrix.unshift item
  end

  result_yaml['Matrix'] = matrix

  File.open('jsontodb-config.yml', 'w') do |file|
    file.write(result_yaml.to_yaml)
  end
end

File.open('config/cache.dat', 'w') do |file|
  file.write(cache)
end
